src/main/java/nl/knaw/huc/resussun/dataextension/TimbuctooExtensionQuery.java

```diff
@@ -0,0 +1,122 @@
+  private final List<String> subjects;
+  private final List<String> predicates;
+
+  public TimbuctooExtensionQuery(String dataSetId, List<String> subjects, List<String> predicates) {
+
+    this.dataSetId = dataSetId;
+    this.subjects = subjects;
+    this.predicates = predicates;
+  }
+
+  public TimbuctooRequest createQuery() {
+    String query = String.format("query extensionQuery {\n" +
+            "  dataSets {\n" +
+            "    %s {\n" +
+            "      %s" +
+            "    }\n" +
+            "  }\n" +
+            "}",
+        dataSetId,
+        createSubjectQuery(subjects, predicates)
+    );
+
+
+    return new TimbuctooRequest(query, Maps.newHashMap());
+  }
+
+  private String createSubjectQuery(List<String> subjects, List<String> predicates) {
+    StringBuilder subjectQuery = new StringBuilder();
+    String predicateQuery = createPredicateQuery(predicates);
+    for (String subject : subjects) {
+      subjectQuery.append(String.format("%s: subject(uri: \"%s\") {\n" +
+          "  %s\n" +
+          "}\n", escapeGraphQl(subject), subject, predicateQuery
+      ));
+    }
+
+    return subjectQuery.toString();
+  }
+
+  private String createPredicateQuery(List<String> predicates) {
+    StringBuilder predicateQuery = new StringBuilder();
+    for (String predicate : predicates) {
+      predicateQuery.append(String.format("%s: getAllOfPredicate(uri: \"%s\" outgoing:true) {\n" +
+          "  values {\n" +
+          "    value\n" +
+          "  }\n" +
+          "}\n", escapeGraphQl(predicate), predicate
+      ));
+    }
+
+    return predicateQuery.toString();
+  }
+
+  private String escapeGraphQl(String label) {
+    return label.replaceAll("[!$().:=@\\[\\]{|}/]", "_");
+  }
+
+
+  public TimbuctooResponseMapper<Map<String, Map<String, List<? extends PropertyValue>>>> createMapper() {
+    return new TimbuctooExtensionQueryResponseMapper(dataSetId, subjects, predicates, this::escapeGraphQl);
+  }
+
+  private static class TimbuctooExtensionQueryResponseMapper
+      implements TimbuctooResponseMapper<Map<String, Map<String, List<? extends PropertyValue>>>> {
+
+    private final String dataSetId;
+    private final List<String> subjects;
+    private final List<String> predicates;
+    private final Function<String, String> escapeGraphQl;
+
+
+    public TimbuctooExtensionQueryResponseMapper(
+        String dataSetId,
+        List<String> subjects,
+        List<String> predicates,
+        Function<String, String> escapeGraphQl
+    ) {
+      this.dataSetId = dataSetId;
+      this.subjects = subjects;
+      this.predicates = predicates;
+      this.escapeGraphQl = escapeGraphQl;
+    }
+
+    @Override
+    public Map<String, Map<String, List<? extends PropertyValue>>> mapResponse(JsonNode timbuctooResponse) {
+      JsonNode data = timbuctooResponse.get("data").get("dataSets").get(dataSetId);
+      Map<String, Map<String, List<? extends PropertyValue>>> mappedResponse = Maps.newHashMap();
```

Comment:
Same comment as before: These Guava functions (newHasMap, newArrayList...) are considered deprecated. Just use `new HashMap<>();` and `new ArrayList<>();` instead.